rmw_fastrtps_shared_cpp/src/rmw_node_info_and_types.cpp

```diff
@@ -85,7 +85,7 @@ rmw_ret_t __get_guid_by_name(
         kLoggerTag,
         "Unable to find GUID for node: %s", node_name);
       RMW_SET_ERROR_MSG("Unable to find GUID for node ");
-      return RMW_RET_ERROR;
+      return RMW_RET_NON_EXISTENT_NODE_NAME;
```

Comment:
Is the caller still expected to clear the error message? Or with the explicit return code the error message could be removed?

Reply:
We usually set the error message when returning an explicit return code, so I will follow that.
Deleting the logging of the error, as is unnecessary.